tests/baselines/reference/inferTypes1.types

```diff
@@ -490,6 +490,6 @@ const result = invoker('test', true)({ test: (a: boolean) => 123 })
 >123 : 123
 
 type Foo2<A extends any[]> = ReturnType<(...args: A) => string>;
->Foo2 : string
+>Foo2 : Foo2<A>

```

Comment:
This change is caused by pretty similar stuff to the one described [here](https://github.com/microsoft/TypeScript/pull/50034/files#r928752433).

The only~ difference here is that we get:
```
source // any[]
target // A
```
and there is not a lot of logic related to handling type params in the `structuredTypeRelatedToWorker`. So this `target` stays as `A`, its constraint is not really requested - but, again, the constraint for some reason is `unknown`, instead of `any[]`. Even though `T`'s declaration points to `T extends any[]`.